pkg/kubelet/status/status_manager.go

```diff
@@ -714,3 +714,41 @@ func NeedToReconcilePodReadiness(pod *v1.Pod) bool {
 	}
 	return false
 }
+
+//SidecarsStatus contains three bools, whether the pod has sidecars, if the all the sidecars are ready and if the non sidecars are in a waiting state.
+type SidecarsStatus struct {
```

Comment:
this seems to be an internal record of kubelet's `kuberuntime_manager` so should not get exported?

Reply:
yeah will change

Reply:
interesting, changing this actually breaks golint,
```
Errors from golint:
pkg/kubelet/status/status_manager.go:727:37: exported func GetSidecarsStatus returns unexported type status.sidecarsStatus, which can be annoying to use
```
So I will leave as is